### ISSUE TITLE
create filtered tables

### ISSUE THREAD

**Issue by dmgerman** [SiRumCz/Researches-on-NPM-public-packages#9]
For each table in the original ghtorrent, extract only the tuples you need.

For example, given that you have a relation with the ids of the NPM projects, I created a table of "commits" that only pertain to these projects:

select timeofday();; create table dn_project_commits as select * from project_commits p join dn_project_id using (project_id); select timeofday();

i also create a table of the project information:

select timeofday();; create table dn_projects as select p.* from projects p join dn_project_id on (project_id=id); select timeofday();

basically, for each table, create a dn_table.

The order in which you do it is important. Now that you have projects and project_commits, do commits, then authors, then forks, etc.

This should reduce the amount of data you need by at least a factor of 10. The project commits table went from 6billion to .3 billion tuples (a reduction of 20). The projects table was reduced much more; almost 2000 times.

Each table will take some time to be created. It took between 2 and 3 hrs per table when I did it.

**Comment by SiRumCz** [comment]
When I am creating filtered table for dn_commits, I keep getting the disk full problem, from what I am seeing from the system monitor is that after I run the query, it consistently consumes all the disk space until it's full. Here is my query:
```
CREATE TABLE dn_commits TABLESPACE nas_wd_red AS 
SELECT commits.* FROM commits
JOIN aug_10_npm_packages
USING (project_id);
```
